packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -173,6 +173,25 @@ export class SelectedNetworkController extends BaseController<
     });
   }
 
+  setNetworkClientIdForDomain(
+    domain: Domain,
+    networkClientId: NetworkClientId,
+  ) {
+    if (!this.state.perDomainNetwork) {
+      Object.entries(this.state.domains).forEach(
+        ([entryDomain, networkClientIdForDomain]) => {
+          if (
+            networkClientIdForDomain !== networkClientId &&
+            entryDomain !== domain
+          ) {
+            this.#setNetworkClientIdForDomain(entryDomain, networkClientId);
+          }
+        },
+      );
+    }
```

Comment:
nit. dropping the `entryDomain !== domain` check and exiting early makes it more clear that you're trying to avoid calling setNetworkClientIdForDomain twice for the passed in domain.

```suggestion
    if (!this.state.perDomainNetwork) {
      Object.entries(this.state.domains).forEach(
        ([entryDomain, networkClientIdForDomain]) => {
          if (
            networkClientIdForDomain !== networkClientId
          ) {
            this.#setNetworkClientIdForDomain(entryDomain, networkClientId);
          }
        },
      );
      return;
    }
```

Reply:
I think that would be different, no? 

The `entryDomain !== domain` check is inside the `Object.entries(this.state.domains).forEach`, which just makes sure that we dont end up in an inifite loop.